src/Simple.OData.Client.UnitTests/Core/TypedFilterAsKeyTests.cs

```diff
@@ -103,6 +103,30 @@ public async Task FindAllByFilterAsKeyEqualAndExtraClause()
             Assert.Equal(string.Format("Products?$filter=ProductID%20eq%201%20and%20ProductName%20eq%20{0}abc{0}", 
                 Uri.EscapeDataString("'")), commandText);
         }
+       
+        [Fact]
+        public async Task FindAllByFilterAsFirstPartOfCompoundKeyEqualAndExtraClause()
+        {
+            var command = _client
+                .For<OrderDetail>()
+                .Filter(x => x.OrderID == 1 && x.Quantity == 1 );
+            var commandText = await command.GetCommandTextAsync();
+
+            var expected = string.Format("Order_Details?$filter=OrderID%20eq%201%20and%20Quantity%20eq%201", Uri.EscapeDataString("'"));
+            Assert.Equal(expected, commandText);
+        }
+
+        [Fact]
+        public async Task FindAllByFilterAsSecondPartOfCompoundKeyEqualAndExtraClause()
+        {
+            var command = _client
+                .For<OrderDetail>()
+                .Filter(x => x.ProductID == 1 && x.Quantity == 1);
+            var commandText = await command.GetCommandTextAsync();
+
+            var expected = string.Format("Order_Details?$filter=ProductID%20eq%201%20and%20Quantity%20eq%201", Uri.EscapeDataString("'"));
```

Comment:
This does not look like it is using the format argument, and can just be a plain string.